cypress/integration/configuration/productTypes.js

```diff
@@ -115,7 +129,147 @@ filterTests({ definedTags: ["all"] }, () => {
           getProductType(productType.id);
         })
         .then(productType => {
-          expect(productType.variantAttributes[0].name).to.eq(startsWith);
+          expect(productType.assignedVariantAttributes[0].attribute.name).to.eq(
+            startsWith
+          );
+        });
+    });
+
+    it("should be able to delete product type. TC: SALEOR_1505", () => {
+      const name = `${startsWith}${faker.datatype.number()}`;
+
+      createTypeProduct({ name, hasVariants: false }).then(productType => {
+        cy.visitAndWaitForProgressBarToDisappear(
+          productTypeDetailsUrl(productType.id)
+        )
+          .get(BUTTON_SELECTORS.deleteButton)
+          .click()
+          .addAliasToGraphRequest("ProductTypeDelete")
+          .get(SHARED_ELEMENTS.warningDialog)
+          .find(BUTTON_SELECTORS.deleteButton)
+          .click()
+          .waitForRequestAndCheckIfNoErrors("@ProductTypeDelete");
+        getProductType(productType.id).should("be.null");
+      });
+    });
+
+    it("should be able to delete product type with assigned product. TC: SALEOR_1509", () => {
+      const name = `${startsWith}${faker.datatype.number()}`;
+      let productType;
+
+      createTypeProduct({ name, hasVariants: false })
+        .then(productTypeResp => {
+          productType = productTypeResp;
+          createProductInChannel({
+            name,
+            channelId: channel.id,
+            categoryId: category.id,
+            productTypeId: productType.id
+          });
+        })
+        .then(({ product }) => {
+          cy.visitAndWaitForProgressBarToDisappear(
+            productTypeDetailsUrl(productType.id)
+          )
+            .get(BUTTON_SELECTORS.deleteButton)
+            .click()
+            .addAliasToGraphRequest("ProductTypeDelete")
+            .get(SHARED_ELEMENTS.warningDialog)
+            .find(BUTTON_SELECTORS.deleteButton)
+            .should("not.be.enabled")
+            .get(BUTTON_SELECTORS.deleteAssignedItemsConsentCheckbox)
+            .click()
+            .get(SHARED_ELEMENTS.warningDialog)
+            .find(BUTTON_SELECTORS.deleteButton)
+            .click()
+            .waitForRequestAndCheckIfNoErrors("@ProductTypeDelete");
+          getProductType(productType.id).should("be.null");
+          getProductDetails(product.id)
+            .its("body.data.product")
+            .should("be.null");
+        });
+    });
+
+    it("should be able to remove variant attribute from product type. TC: SALEOR_1506", () => {
```

Comment:
This case or 1507 sometime is failing.
![image](https://user-images.githubusercontent.com/29308910/152168569-47672286-1b03-4e70-9b75-bf85ce06830f.png)